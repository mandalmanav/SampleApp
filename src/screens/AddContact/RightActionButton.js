import React, { useDebugValue } from 'react';
import { Pressable ,Alert} from 'react-native';
import { StyleSheet, Text, View, Button } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { addContactAction } from '../../store/actions/contacts';

export default ({navigation }) => {
    let {contact} = useSelector(state=>state)
    let dispatch = useDispatch()
    return (<Pressable onPress={async() => {
        if (contact.firstName==undefined||!contact.firstName.trim().length) {
                        Alert.alert(
                            "Invalid",
                            "First name cannot be left blank",
                            [
                                {
                                    text: "OK",
                                    onPress: () => {
                                        return
                                    },
                                    style: "cancel"
                                }
                            ]
                        );
                        return
                    }
                    if (contact.lastName==undefined||!contact.lastName.trim().length) {
                        Alert.alert(
                            "Invalid",
                            "Last name cannot be left blank",
                            [
                                {
                                    text: "OK",
                                    onPress: () => {
                                        return
                                    },
                                    style: "cancel"
                                }
                            ]
                        );
                        return
                    }
                    if (contact.mobile==undefined||String(contact.mobile).length < 10) {
                        Alert.alert(
                            "Invalid",
                            "Mobile number should consist of 10 digits",
                            [
                                {
                                    text: "OK",
                                    onPress: () => {
                                        return
                                    },
                                    style: "cancel"
                                }
                            ]
                        );
                        return
                    }
        await addContactAction(dispatch,contact)
        navigation.navigate('ContactDetails')
      }}><Text style={{ ...styles.primaryText, ...styles.actionFS, ...{ marginRight: 8 } }}>Save</Text>
      </Pressable>)
}

const styles = StyleSheet.create({
    primaryText: {
        color: '#4d90fe'
    },
    actionFS: {
        fontSize: 16
    }
});