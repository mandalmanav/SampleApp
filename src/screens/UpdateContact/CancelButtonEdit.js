import React, { useDebugValue } from 'react';
import { Pressable } from 'react-native';
import { StyleSheet, Text, View, Button } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { addContactAction, updateContactAction } from '../../store/actions/contacts';
import { SET_SELECTED } from '../../store/constants';

export default ({ navigation }) => {
    let { contact, contacts } = useSelector(state => state)
    let dispatch = useDispatch()
    return (<Pressable onPress={async () => {
        let oldContact = contacts.filter(c => c.id == contact.id)[0]
        dispatch({
            type: SET_SELECTED,
            payload: oldContact
        })

        navigation.navigate('ContactDetails')
    }}><Text style={{ ...styles.primaryText, ...styles.actionFS, ...{ marginLeft: 8 } }}>Cancel</Text>
    </Pressable>)
}

const styles = StyleSheet.create({
    primaryText: {
        color: '#4d90fe'
    },
    actionFS: {
        fontSize: 16
    }
});