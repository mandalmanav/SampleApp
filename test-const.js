export const CONTACTS=[{
    firstName:'Manav',
    lastName:'Mandal',
    image:'',
    mobile:'9999999999',
    id:'q',
    countryCode:'+91'
},{
    firstName:'Aman',
    lastName:'Vatsa',
    image:'',
    mobile:'9999999999',
    id:'r',
    countryCode:'+91'
}]

export const SEARCH_TERM='am'