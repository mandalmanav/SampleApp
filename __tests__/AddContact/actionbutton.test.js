
import { render, fireEvent } from '@testing-library/react-native';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import React from 'react';
import { CONTACTS, SEARCH_TERM } from '../../test-const';
import RightActionButton from '../../src/screens/AddContact/RightActionButton';
import RightActionButtonEdit from '../../src/screens/UpdateContact/RightActionButtonEdit';
import CancelButtonEdit from '../../src/screens/UpdateContact/CancelButtonEdit';

describe('Test Action Buttons', () => {
    let initialState = { contacts: CONTACTS, searchTerm: '', contact: {} }
    const mockStore = configureStore()
    let store, wrapper
    store = mockStore(initialState)
    it('Match Snapshot save button in add screen', () => {
        const component = render(<Provider store={store}><RightActionButton /></Provider>)
        expect(component).toMatchSnapshot()
    })
    it('Check for text', () => {
        const {getByText} = render(<Provider store={store}><RightActionButton /></Provider>)
        expect(getByText('Save')).not.toBeNull()
    })
    it('Match Snapshot save button in edit screen', () => {
        const component = render(<Provider store={store}><RightActionButtonEdit /></Provider>)
        expect(component).toMatchSnapshot()
    })
    it('Check for text', () => {
        const {getByText} = render(<Provider store={store}><RightActionButtonEdit /></Provider>)
        expect(getByText('Save')).not.toBeNull()
    })
    it('Match Snapshot cancel button in edit screen', () => {
        const component = render(<Provider store={store}><CancelButtonEdit /></Provider>)
        expect(component).toMatchSnapshot()
    })
    it('Check for text', () => {
        const {getByText} = render(<Provider store={store}><CancelButtonEdit /></Provider>)
        expect(getByText('Cancel')).not.toBeNull()
    })
})