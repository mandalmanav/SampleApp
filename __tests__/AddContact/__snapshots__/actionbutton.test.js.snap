// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Action Buttons Match Snapshot cancel button in edit screen 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      Object {
        "color": "#4d90fe",
        "fontSize": 16,
        "marginLeft": 8,
      }
    }
  >
    Cancel
  </Text>
</View>
`;

exports[`Test Action Buttons Match Snapshot save button in add screen 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      Object {
        "color": "#4d90fe",
        "fontSize": 16,
        "marginRight": 8,
      }
    }
  >
    Save
  </Text>
</View>
`;

exports[`Test Action Buttons Match Snapshot save button in edit screen 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      Object {
        "color": "#4d90fe",
        "fontSize": 16,
        "marginRight": 8,
      }
    }
  >
    Save
  </Text>
</View>
`;
